Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -526,6 +555,144 @@ BTDiagnostics::InitializeFieldFunctors (int lev)
+                                       m_cell_centered_data[lev].get(), lev,
+                                       nvars, m_num_buffers, m_varnames);
+    }
+
+    // Reset field functors for cell-center multifab
+    m_cell_center_functors[lev].clear();
+    m_cell_center_functors[lev].resize(m_cellcenter_varnames_fields.size());
+
+    for (int comp=0, n=m_cell_center_functors.at(lev).size(); comp<n; comp++){
+        if        ( m_cellcenter_varnames_fields[comp] == "Er" ){
+            m_cell_center_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_Efield_aux(lev, 0), lev, m_crse_ratio, false, ncomp);
+        } else if ( m_cellcenter_varnames_fields[comp] == "Et" ){
+            m_cell_center_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_Efield_aux(lev, 1), lev, m_crse_ratio, false, ncomp);
+        } else if ( m_cellcenter_varnames_fields[comp] == "Ez" ){
+            m_cell_center_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_Efield_aux(lev, 2), lev, m_crse_ratio, false, ncomp);
+        } else if ( m_cellcenter_varnames_fields[comp] == "Br" ){
+            m_cell_center_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_Bfield_aux(lev, 0), lev, m_crse_ratio, false, ncomp);
+        } else if ( m_cellcenter_varnames_fields[comp] == "Bt" ){
+            m_cell_center_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_Bfield_aux(lev, 1), lev, m_crse_ratio, false, ncomp);
+        } else if ( m_cellcenter_varnames_fields[comp] == "Bz" ){
+            m_cell_center_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_Bfield_aux(lev, 2), lev, m_crse_ratio, false, ncomp);
+        } else if ( m_cellcenter_varnames_fields[comp] == "jr" ){
+            m_cell_center_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_current_fp(lev, 0), lev, m_crse_ratio, false, ncomp);
+        } else if ( m_cellcenter_varnames_fields[comp] == "jt" ){
+            m_cell_center_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_current_fp(lev, 1), lev, m_crse_ratio, false, ncomp);
+        } else if ( m_cellcenter_varnames_fields[comp] == "jz" ){
+            m_cell_center_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_current_fp(lev, 2), lev, m_crse_ratio, false, ncomp);
+        } else if ( m_cellcenter_varnames_fields[comp] == "rho" ){
+            m_cell_center_functors[lev][comp] = std::make_unique<RhoFunctor>(lev, m_crse_ratio, -1, false, ncomp);
+        }
+    }
+
+#endif
+    amrex::ignore_unused(lev);
+}
+
+void
+BTDiagnostics::AddRZModesToOutputNames (const std::string& field, const int ncomp, bool cellcenter_data)
+{
+#ifdef WARPX_DIM_RZ
+    // In cylindrical geometry, real and imag part of each mode are also
+    // dumped to file separately, so they need to be added to m_varnames
+    if (cellcenter_data) {
```

Comment:
Just some explicit naming of terms to make it easy to follow
```suggestion
    // we number modes from 0 to (nmodes-1);
    // mode 0 is purely real, all higher modes are complex
    int const nmodes = (ncomp+1)/2;

    if (cellcenter_data) {
```